### PR TITLE
Add more SMS functionality

### PR DESCRIPTION
**Pull request by sethvargo**
/assign @mikehelmick

### CODE REVIEW

File: pkg/controller/issueapi/issue.go [hunk 1]
```diff
@@ -84,27 +99,28 @@ func (ic *IssueAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if request.SymptomDate != "" {
 		if parsed, err := time.Parse("2006-01-02", request.SymptomDate); err != nil {
 			ic.logger.Errorf("time.Parse: %v", err)
-			controller.WriteJSON(w, http.StatusOK, api.Error("invalid test date: %v", err))
+			controller.WriteJSON(w, http.StatusUnprocessableEntity, api.Error("invalid test date: %v", err))
```
Comment (mikehelmick) [inline pkg/controller/issueapi/issue.go:102]: change error text to symptom onset date - I missed that when we changed the date field

File: pkg/controller/issueapi/issue.go [hunk 2]
```diff
@@ -123,10 +139,19 @@ func (ic *IssueAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
-		controller.WriteJSON(w, http.StatusOK, api.Error("error generating verification, wait a moment and try again"))
+		controller.WriteJSON(w, http.StatusInternalServerError, api.Error("error generating verification, wait a moment and try again"))
 		return
 	}
 
+	if smsProvider != nil {
+		message := fmt.Sprintf(smsTemplate, code, int(expiration.Minutes()))
+		if err := smsProvider.SendSMS(ctx, request.Phone, message); err != nil {
+			ic.logger.Errorf("otp.SendSMS: %v", err)
+			controller.WriteJSON(w, http.StatusInternalServerError, api.Error("failed to send sms: %v", err))
```
Comment (mikehelmick) [inline pkg/controller/issueapi/issue.go:150]: Do we want to delete the code in that case?

File: pkg/controller/issueapi/issue.go [hunk 3]
```diff
@@ -156,5 +181,8 @@ func (ic *IssueAPI) getAuthorizationFromContext(r *http.Request) (*database.Auth
 		return nil, nil, fmt.Errorf("unable to identify authorized requestor")
 	}
 	return nil, user, nil
-
 }
+
+const smsTemplate = `Your exposure notifications verification code is %s. ` +
```
Comment (mikehelmick) [inline pkg/controller/issueapi/issue.go:186]: this is fine for now, we'll probably need PHA customization here

and we'll need a URI to the app at some point.

File: pkg/controller/issueapi/issue.go [hunk 4]
```diff
@@ -68,11 +68,26 @@ func (ic *IssueAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Verify SMS configuration
+	smsProvider, err := ic.db.GetSMSProvider("") // TODO(sethvargo): pass in realm
```
Comment (mikehelmick) [inline pkg/controller/issueapi/issue.go:72]: would be good to put a write through cache in front of these (can be future PR)

File: pkg/controller/issueapi/issue.go [hunk 5]
```diff
@@ -68,11 +68,26 @@ func (ic *IssueAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	if err != nil {
+		ic.logger.Errorf("otp.GetSMSProvider: %v", err)
+		controller.WriteJSON(w, http.StatusInternalServerError, api.Error("failed to get sms provider"))
+		return
+	}
+
+	// Phone number is required if SMS is configured
+	if smsProvider != nil && request.Phone == "" {
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply pkg/controller/issueapi/issue.go]: Done.

File: pkg/controller/issueapi/issue.go (hunk 2 above)
Comment (sethvargo) [reply pkg/controller/issueapi/issue.go:153]: I think we want to save the code and present it on the screen for the clinician to read. WDYT?

File: pkg/controller/issueapi/issue.go (hunk 3 above)
Comment (sethvargo) [reply pkg/controller/issueapi/issue.go:189]: Yea I figured we'd iterate on it.

File: cmd/server/assets/home.html [hunk 6]
```diff
@@ -60,6 +60,21 @@ <h1>Issue Diagnosis Verification Code</h1>
...
 
+      {{ if .smsEnabled }}
+      <div class="row form-group">
+        <label for="symptomDate" class="col-sm-3">Patient phone</label>
+        <div class="col-sm-9">
+          <div class="datepicker date input-group p-0 shadow-sm">
+            <input type="tel" id="phone" name="phone" class="form-control" />
+          </div>
+          <small class="form-text text-muted">
+            This must be a phone number capable of receiving SMS text messages.
```
[the remark this answers is not in the document]
Comment (sethvargo) [reply cmd/server/assets/home.html]: So, Twilio is "smart" in that by default it uses the country code associated with the account if one isn't provided. I'm thinking we leave this as-is for now and maybe iterate on it later? I originally went down a path of some client-side javascript stuff, but the twilio api error responses are pretty good.

File: pkg/controller/issueapi/issue.go (hunk 4 above)
Comment (sethvargo) [reply pkg/controller/issueapi/issue.go]: Done

File: pkg/controller/issueapi/issue.go (hunk 1 above)
Comment (sethvargo) [reply pkg/controller/issueapi/issue.go]: Done

File: pkg/controller/issueapi/issue.go (hunk 2 above)
Comment (sethvargo) [reply pkg/controller/issueapi/issue.go:159]: Done

Comment (mikehelmick) [reply pkg/controller/issueapi/issue.go:159]: yeah, I think it's fine, but this only sends the error back

**Pull request merged**